src/main/java/com/laytonsmith/core/events/drivers/PlayerEvents.java

```diff
@@ -910,6 +910,67 @@ public boolean matches(Map<String, Mixed> prefilter, BindableEvent e) throws Pre
 			return false;
 		}
 
+		@Override
+		public PrefilterBuilder getPrefilters() {
+			return PrefilterBuilder
+					.start("button", PrefilterType.STRING_MATCH, new PrefilterMatcher() {
+						@Override
+						public String getStringParameter(BindableEvent e) {
+							if(e instanceof MCPlayerInteractEvent pie) {
+								if(pie.getAction().equals(MCAction.LEFT_CLICK_AIR)
+									|| pie.getAction().equals(MCAction.LEFT_CLICK_BLOCK)) {
+									return "left";
+								}
+								if(pie.getAction().equals(MCAction.RIGHT_CLICK_AIR)
+										|| pie.getAction().equals(MCAction.RIGHT_CLICK_BLOCK)) {
+									return "right";
+								}
+							}
+							return null;
+						}
+					})
+					.set("itemname", PrefilterType.STRING_MATCH, new PrefilterMatcher() {
+						@Override
+						public String getStringParameter(BindableEvent e) {
+							if(e instanceof MCPlayerInteractEvent pie) {
+								return pie.getItem().getType().getName();
```

Comment:
Item can be null in addition to "block" type.

Reply:
Need to give this some thinks too.

Reply:
At least for String matches it should be as easy as returning a String "null". However, other match types might not be so fortunate. I'm not sure if there are any others that support null that aren't strings.